tools/provision/chocolatey/zstd.ps1

```diff
@@ -0,0 +1,76 @@
+#  Copyright (c) 2014-present, Facebook, Inc.
+#  All rights reserved.
+#
+#  This source code is licensed under the BSD-style license found in the
+#  LICENSE file in the root directory of this source tree. An additional grant
+#  of patent rights can be found in the PATENTS file in the same directory.
+
+# Update-able metadata
+$version = '1.2.0'
+$chocoVersion = '1.2.0-r1'
+$packageName = 'zstd'
+$projectSource = 'https://github.com/facebook/zstd'
+$packageSourceUrl = 'https://github.com/facebook/zstd'
+$authors = 'Facebook'
+$owners = 'Facebook'
+$copyright = 'https://github.com/facebook/zstd/blob/master/LICENSE'
+$license = 'https://github.com/facebook/zstd/blob/master/LICENSE'
+$url = "https://github.com/facebook/zstd/archive/v$version.zip"
+
+# Invoke our utilities file
+. "$(Split-Path -Parent $MyInvocation.MyCommand.Definition)\osquery_utils.ps1"
+
+# Invoke the MSVC developer tools/env
+Invoke-BatchFile "$env:VS140COMNTOOLS\..\..\vc\vcvarsall.bat" amd64
+
+# Time our execution
+$sw = [System.Diagnostics.StopWatch]::startnew()
+
+# Keep the location of build script, to bring with in the chocolatey package
+$buildScript = $MyInvocation.MyCommand.Definition
+
+# Create the choco build dir if needed
+$buildPath = Get-OsqueryBuildPath
+if ($buildPath -eq '') {
+  Write-Host '[-] Failed to find source root' -foregroundcolor red
+  exit
+}
+$chocoBuildPath = "$buildPath\chocolatey\$packageName"
+if (-not (Test-Path "$chocoBuildPath")) {
+  New-Item -Force -ItemType Directory -Path "$chocoBuildPath"
+}
+Set-Location $chocoBuildPath
+
+# Retreive the source
+Invoke-WebRequest $url -OutFile "$packageName-$version.zip"
+
+# Extract the source
+7z x "$packageName-$version.zip"
+$sourceDir = "$packageName-$version/build"
+Set-Location $sourceDir
+.\VS_scripts\build.generic.cmd VS2015 x64 Release v140
+
+Set-Location '..'
+
+# Construct the Chocolatey Package
+$chocoDir = New-Item -ItemType Directory -Path 'osquery-choco'
+Set-Location $chocoDir
+$includeDir = New-Item -ItemType Directory -Path 'local\include'
+$libDir = New-Item -ItemType Directory -Path 'local\lib'
+$srcDir = New-Item -ItemType Directory -Path 'local\src'
+
+Write-NuSpec $packageName $chocoVersion $authors $owners $projectSource $packageSourceUrl $copyright $license
+Set-Location '..'
+Copy-Item "build\VS_scripts\bin\Release\x64\libzstd.lib" $libDir
```

Comment:
Once you've made the above changes some of these paths _might_ need changing. I try to keep as many of the build artifacts as possible in a subfolder `osquery-win-build` of the `$sourceDir` as I can, and then when copy/pasting libs/headers one can `Copy-Item "$sourceDir\$buildDir\libs" "$sourceDir\$chocoDir\lib`.. type of thing. This takes a little playing around with manually building the libs to get a feel for the directory structure.